### PR TITLE
Update support and compatibitility sections of documentation

### PR DESCRIPTION
**Pull request by palnabarun**
- Refactor the format of compatibility matrix
- Clarify support cycles for the releases temporarily
- Explain the version bump and misses in between v12 and v17.

Signed-off-by: Nabarun Pal <pal.nabarun95@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +131,8 @@ Kubernetes supports [three minor releases](https://github.com/kubernetes/communi
 
 Note: There would be no maintenance for alpha/beta releases except the latest one.
 
+**Exception to the above support rule:** Since we are running behind on releases, we will be supported Alpha/Beta releases for a greater number of clients until we catch up with the upstream version.
```
Comment (yliaog) [inline README.md:134]: s/will be supported/will support/

File: README.md [hunk 2]
```diff
@@ -80,14 +80,12 @@ All APIs and Models' documentation can be found at the [Generated client's READM
...
-|                    | Kubernetes 1.13 | Kubernetes 1.14 | Kubernetes 1.15 |
-|--------------------|-----------------|-----------------|-----------------|
-| client-python 9.0  |✓                |+-               |+-               |
-| client-python 10.0 |+-               |✓                |+-               |
-| client-python 11.0 |+-               |+-               |✓                |
-| client-python HEAD |+-               |+-               |+-               |
+- [client 9.y.z](https://pypi.org/project/kubernetes/9.0.1/): Kubernetes 1.12 or below (+-), Kubernetes 1.13 (✓), Kubernetes 1.14 or above (+-)
+- [client 10.y.z](https://pypi.org/project/kubernetes/10.1.0/): Kubernetes 1.13 or below (+-), Kubernetes 1.14 (✓), Kubernetes 1.14 or above (+-)
+- [client 11.y.z](https://pypi.org/project/kubernetes/11.0.0/): Kubernetes 1.14 or below (+-), Kubernetes 1.15 (✓), Kubernetes 1.16 or above (+-)
+- [client 12.y.z](https://pypi.org/project/kubernetes/12.0.1/): Kubernetes 1.15 or below (+-), Kubernetes 1.16 (✓), Kubernetes 1.17 or above (+-)
```
Comment (roycaihw) [inline README.md:88]: v17.14.0a1 is uploaded to pypi. Could you add it to the list?

Comment (roycaihw) [reply README.md:88]: also could you add a paragraph explaining the different meanings of the version numbers pre/post v17?

File: README.md (hunk 1 above)
Comment (palnabarun) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (palnabarun) [reply README.md:88]: Both of them incorporated.

**Pull request merged**